lib/oppen/mixins.rb

```diff
@@ -21,24 +21,47 @@ def upsize_circular_array(arr, offset)
     #
     # @param tokens [Array[Token]]
     # @param base_indent [Integer]
+    # @param printer_name [String]
     #
     # @return [String]
-    def tokens_to_wadler(tokens, base_indent = 4)
+    def tokens_to_wadler(tokens, base_indent: 0, printer_name: 'out')
+      nb_spaces = base_indent
       out = StringIO.new
-      write = ->(txt, nb_spaces) {
-        out.write("#{' ' * nb_spaces}#{txt}\n")
+
+      write = ->(txt, indentation) {
+        out.write("#{' ' * indentation}#{txt}\n")
       }
-      nb_spaces = base_indent
+      dump = ->(str) {
+        str.dump.gsub(/\\u(\h{4})/) { Regexp.last_match(1).to_i(16).chr(Encoding::UTF_8) }
+      }
+      display_break_token = ->(token) {
+        if token.offset.positive?
+          write.call("#{printer_name}.nest(#{token.offset}, \"\", \"\") {", nb_spaces)
+          nb_spaces += 2
+        end
+
+        case token
+        in Token::LineBreak
+          write.call("#{printer_name}.break(line_continuation: #{dump.call(token.line_continuation)})", nb_spaces)
+        in Token::Break
+          write.call("#{printer_name}.breakable(#{dump.call(token.str)}, width: #{token.width}, " \
```

Comment:
I prefer using the `.()` syntax for calls:

```suggestion
          write.("#{printer_name}.breakable(#{dump.call(token.str)}, width: #{token.width}, " \
```

You could also do `write[...]` but I don't like it.